docs/concepts/node.rst

```diff
@@ -0,0 +1,94 @@
+
+Nodes are simply Python functions
+---------------------------------
+Nodes are defined using the native Python function definition API. The figure below shows how the functions ``A()``, ``B()``, and ``C()`` are automatically assembled into a dataflow. Notice that function definitions don't have any special "Hamilton code". Also, the function name and return type ``A() -> int`` match the parameter ``A: int`` found in functions ``B()`` and ``C()``.
+
+.. image:: ../_static/abc.png
+    :scale: 50
+    :align: center
+
+Anatomy of a node
+-----------------
+The following figure and table detail how a Python function maps to a Hamilton node.
+
+
+.. image:: ../_static/function_anatomy.png
+    :scale: 18%
+    :align: center
+
+
+.. list-table::
+   :header-rows: 1
+
+   * - id
+     - Function components
+     - Node components
+   * - 1
+     - Function name and return type annotation
+     - Node name and type
+   * - 2
+     - Parameter(s) name and type annotation
+     - Node dependencies
+   * - 3
+     - Docstring
+     - Description of the node return value
+   * - 4
+     - Function body
+     - Implementation of the node
+
+.. note::
+  A `node` is an object Hamilton creates and uses as part of a `dataflow`. A `function` is a piece of Python code wrote by someone. A `function` almost always map to a single `node` so the two terms are often used interchangeably.
+
+  However, Hamilton allows to create multiple nodes with a single function to keep your code `DRY <https://en.wikipedia.org/wiki/Don't_repeat_yourself>`_. We'll discuss these features later in this guide.
+
+Function naming
+---------------
+Hamilton strongly agrees with the `Zen of Python <https://peps.python.org/pep-0020/>`_ #2: "Explicit is better than implicit". Meaningful function names help document what functions do, so don't shy away from longer names. If you were to come across a function named ``life_time_value`` versus ``ltv`` versus ``l_t_v``, which one is most obvious? Remember your code usually lives a lot longer that you ever think it will.
+
+Unlike the common practice of including meaningful verbs in function names (e.g., ``get_credentials()``, ``statistical_test()``), with Hamilton, the function name should more closely align with nouns. That's because the function name determines the node name and how data will be queried. Therefore, names that describe the node result rather than its action may be more readable (e.g., ``credentials()``, ``statistical_results()``).
+
+
+Node definitions -> Dataflow definition
+---------------------------------------
+Defining an Hamilton node requires specifying it's dependencies as function parameters. By walking through dependencies, Hamilton can automatically build a dataflow from a collection of node definitions. In practice, Hamilton loads all functions of a Python modules, so you don't have to import them one by one.
```

Comment:
```suggestion
A function's parameters declare dependencies, which in turn form edges between nodes, to link a dataflow together. So the steps to create a dataflow are:

1. Provide modules that contain functions for Hamilton to parse.
2. For each function, Hamilton creates a node(s).
3. Based on each function's parameters, edges are created between Hamilton nodes.
4. At the end of this process we have a dataflow defined.

Note: Hamilton automatically crawls modules provided for functions so you don't have to import them one by one or explicitly define what the dataflow is, because you've already defined it via function naming and parameters.
```